### PR TITLE
THU/table_edition

### PR DESCRIPTION
**Pull request by csm-thu**

### CODE REVIEW

**Comment by csm-thu** [comment]
Remaining work: write .mdx documentation

File: src/inputs/Table/ColumnTypes.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    valueSetter: _editModeSetter,
+    onCellValueChanged: (event) => {
+      onCellChange(event);
+    },
+  };
+};
+
+export const getColumnTypes = (dateFormat) => {
+  const _dateComparator = (valueA, valueB, nodeA, nodeB, isInverted) => {
+    return DateUtils.parse(valueA, dateFormat) > DateUtils.parse(valueB, dateFormat) ? 1 : -1;
```
Comment (COSMO-VMI) [inline src/inputs/Table/ColumnTypes.js:119]: Why do you return `-1` and `1`, instead of `true` and `false`?

Comment (csm-thu) [reply src/inputs/Table/ColumnTypes.js:119]: Comparison functions usually return -1, 0 and 1 to describe the 3 possible results of the comparison : less than, equal to or greater than.
In this case, the comparison function is given to AgGrid, and it expects -1, 0 and 1 as returned values.

Comment (COSMO-VMI) [reply src/inputs/Table/ColumnTypes.js:119]: OK, you didn't handle the 0 case, so I was wondering. Thanks.

**Review by nborde-CSM — COMMENTED** [review]
Some little remarks but LGTM

**Review by COSMO-VMI — APPROVED** [review]
LGTM, GG

File: src/inputs/Table/ColumnTypes.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  }
+  // Force date format before setting it
+  params.data[params.colDef.field] = DateUtils.format(DateUtils.parse(newValue, dateFormat), dateFormat);
+  return true;
+};
+
+const _intSetter = (params) => {
+  let newValue = parseInt(params.newValue);
+
+  if (!params.context.editMode || isNaN(newValue) || !Number.isSafeInteger(newValue)) {
```
[the remark this answers is not in the document]
Comment (csm-thu) [reply src/inputs/Table/ColumnTypes.js:53]: `isSafeInteger` catches the `NaN` values indeed, I'll simplify the code, thanks :+1:

**Pull request merged**